ocs_ci/ocs/ui/workload_ui.py

```diff
@@ -0,0 +1,492 @@
+        depl_name (str): Name of the deployment to be created. Defaults to None.
+        pvc_name (str): Name of the PVC to be attached by the busybox container. Defaults to None.
+
+        Returns:
+            tuple: The name of the node where the busybox container is deployed, and a deployment name if deployed,
+            otherwise None.
+        """
+        if not node:
+            node = random.choice(get_worker_nodes())
+            logger.info(f"selected node {node} for deployment")
+
+        with tempfile.NamedTemporaryFile(
+            mode="w", suffix=".yaml", delete=True
+        ) as temp_file:
+
+            bb_dict = copy.deepcopy(self.busy_box_depl)
+
+            bb_dict["spec"]["template"]["spec"]["nodeName"] = node
+            bb_dict["metadata"]["namespace"] = namespace
+            if depl_name:
+                bb_dict["metadata"]["name"] = depl_name
+            if pvc_name:
+                bb_dict["spec"]["template"]["spec"].setdefault("volumes", [])
+                bb_dict["spec"]["template"]["spec"]["volumes"].append(
+                    {
+                        "name": "unique-pvc",
+                        "persistentVolumeClaim": {"claimName": pvc_name},
+                    }
+                )
+                bb_dict["spec"]["template"]["spec"]["containers"][0].setdefault(
+                    "volumeMounts", []
+                )
+                bb_dict["spec"]["template"]["spec"]["containers"][0][
+                    "volumeMounts"
+                ].append({"name": "unique-pvc", "mountPath": "/tmp"})
+
+            yaml.dump(bb_dict, temp_file, default_flow_style=False)
+            temp_file.flush()
+            occli = OCP()
+            occli.apply(temp_file.name)
+
+        busy_box_scaled_up = self.wait_busy_box_scaled_up(60, depl_name, namespace)
+        deployment = Deployment(
+            **OCP(
+                kind="deployment",
+                namespace=namespace,
+                resource_name=depl_name,
+            ).get()
+        )
+        self.deployment_list.append(deployment)
+        return node, deployment if busy_box_scaled_up else None
+
+    def wait_busy_box_scaled_up(
+        self, timeout: int, depl_name: str, namespace: str
+    ) -> bool:
+        """
+        Waits for the 'busybox' deployment to be scaled up within the specified timeout.
+
+        Args:
+            timeout (int): The maximum time to wait for the deployment to be scaled up, in seconds.
+            depl_name (str): Name of the deployment to be scaled up.
+            namespace (str): Namespace where the deployment is located.
+        Returns:
+            bool: True if the deployment is successfully scaled up within the timeout, False otherwise.
+
+        """
+        busy_box_ocp_inst = OCP(
+            kind="deployment",
+            namespace=namespace,
+            resource_name=depl_name,
+        )
+        start_time = time.perf_counter()
```

Comment:
you can use TimeoutSampler generator.. https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/utility/utils.py#L1221

Reply:
replaced with TimeoutSampler